### DISCUSSION TITLE
Perfect Link Images

### DISCUSSION THREAD

**Discussion by WhatAUsernameIHave** [Deijin27/RanseiLink#49] [Q&A]
From what is currently known, is there any way to change which PL sprites are used for a warlord? I know that now, the only way to do that is by either swapping Pokemon slots or Warrior slots, but was curious if I might be missing something!

**Comment by Deijin27** [comment]
Could you explain more what you know of how this works please. What is a Perfect Link sprite, when does it show up, is it linked to warrior or pokemon, etc.

There's these sprites where there are multiple emotions for each pokemon, that show up in conversations. But I haven't figured out what decides them yet. I was thinking it might be in the event scripts, but they're pretty difficult to figure out.:
![picture of eevee](https://user-images.githubusercontent.com/40903783/224515145-c1441217-9d88-4e1b-ac4c-b96137f185f5.png)

**Reply by WhatAUsernameIHave** [reply]
Sorry, should've been a bit more clear. When I refer to the Perfect Link sprites I am talking about the conversation sprites. The only thing I got out of my testing is that the selection of sprites it chooses from is linked to a Warlord's and Pokemon's slot. I believe that each emotion is tied between the sprites (for example if Oichi is set to her 38 / Sad sprite, whatever is in Jiggly/Wiggly's slot that matches will change with it). The only thing I'm currently certain about and can modify with RanseiLink is the Warlord ID/Slot which determines the conversation sprites used.

**Reply by Deijin27** [reply]
It must be tied to the pokemon right? So changing oichi to have an eevee shows eevees sprite? (I can't remember if I tested this)

Assuming it is linked to the pokemon id. I don't see anywhere in the pokemon's data it can fit with the remaining unknowns.
Theres 82 unique pokemon with 1 or more of this sprite type, so for a direct identifier we need 7+ bits which we don't have.
It could be that each pokemon stores how many of, or which emotions they have. There's 5 emotions which could fit, but they just don't line up looking at the data.

There is also the mysterious section 2 of the pokemon file, between the pokemon data, and the evolution listings. But I'm pretty sure I've completely wiped this section before, and played up to greenleaf with no obvious differences. This section also has a visible pattern that divides it into 16 sections, so it wouldn't be a fit for the information we're looking for.

It certainly seems like the kind of thing that would be data-driven, but it doesn't seem to be in the pokemon file at least.

**Reply by WhatAUsernameIHave** [reply]
Changing the PL to Eevee doesn’t change it. The only way Oichi will have a Pokémon join her in conversation sprites is if she has whatever is in the Jigglypuff/Wigglytuff ID slots. So it most likely has something to do with both Warrior and Pokémon ID. Unfortunately we just need to find where that data links.
